### PR TITLE
SDK - Lightweight - Convert the names of file inputs and outputs

### PR DESCRIPTION
**Pull request by Ark-kun**
Removing the "_path" and "_file" suffixes from the names of file inputs and outputs.
Problem: When accepting file inputs (outputs), the function inside the component receives file paths (or file streams), so it's natural to call the function parameter "something_file_path" (e.g. model_file_path or number_file_path).
But from the outside perspective, there are no files or paths - the actual data objects (or references to them) are passed in.
It looks very strange when argument passing code looks like this: `component(number_file_path=42)`. This looks like an error since 42 is not a path. It's not even a string.
It's much more natural to strip the names of file inputs and outputs of "_file" or "_path" suffixes. Then the argument passing code will look natural: "component(number=42)"

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/2260)

### CODE REVIEW

File: sdk/python/kfp/components/_python_op.py [hunk 1]
```diff
@@ -38,6 +39,9 @@
 T = TypeVar('T')
 
 
+_FEATURE_STRIP_FILE_IO_NAME_PARTS = True
```
Comment (numerology) [inline sdk/python/kfp/components/_python_op.py]: nit: Maybe a one line comment documenting its functions?

Also, is there any case that we don't want strip out the suffix?

Comment (Ark-kun) [reply sdk/python/kfp/components/_python_op.py]: >is there any case that we don't want strip out the suffix?

There should not be. I've removed the switch.

**Pull request merged**